pkg/apis/apis/v1alpha1/permissionclaims/labels.go

```diff
@@ -37,3 +37,50 @@ func ToLabelKeyAndValue(permissionClaim apisv1alpha1.PermissionClaim) (string, s
 	labelKeyHashLength := validation.LabelValueMaxLength - len(apisv1alpha1.APIExportPermissionClaimLabelPrefix)
 	return apisv1alpha1.APIExportPermissionClaimLabelPrefix + hash[0:labelKeyHashLength], hash, nil
 }
+
+func ValidateClaim(group, resource, identityHash string, grsToBoundResource map[apisv1alpha1.GroupResource]apisv1alpha1.BoundAPIResource) (bool, error) {
+	boundResource, bound := grsToBoundResource[apisv1alpha1.GroupResource{Group: group, Resource: resource}]
+	// This  says that if the resource is not bound, but there is no identity hash, we consider it valid
+	// TODO add ability to check that the GR is knowable by the inernalschema's once added
+	if !bound && identityHash == "" {
+		return true, nil
+	}
+	if !bound && identityHash != "" {
+		return false, fmt.Errorf("unable to find bound resource for %v.%v", resource, group)
+	}
+	if boundResource.Schema.IdentityHash != identityHash {
+		return false, fmt.Errorf("bound resource does not match for %v.%v", resource, group)
+	}
+	return true, nil
+}
+
+func AllClaimLabels(group, resource string, bindings []*apisv1alpha1.APIBinding) (map[string]string, error) {
+	grsToBoundResource := map[apisv1alpha1.GroupResource]apisv1alpha1.BoundAPIResource{}
+	for _, binding := range bindings {
+		for _, resource := range binding.Status.BoundResources {
+			grsToBoundResource[apisv1alpha1.GroupResource{Group: resource.Group, Resource: resource.Resource}] = resource
+		}
+	}
+
+	// add labels for new claims
+	labels := map[string]string{}
+	for _, binding := range bindings {
+		for _, pc := range binding.Spec.AcceptedPermissionClaims {
```

Comment:
🔴  this I commented before: I think this is not correct. Accepted claims can be more than actual claims in the export. We need the union.

Reply:
Just to be clear, this is a new thing, right? This was never the case previously and does require that $SOMETHING is handling the API export -> API binding information. 


We are designing this API now. I don't see why we hold up just a refactor for a new requirement.

Reply:
We discussed this in Slack. This change will come in a follow-up.